csharp/device/Microsoft.Azure.Devices.Client/Message.cs

```diff
@@ -144,7 +178,11 @@ public string To
         {
             get
             {
+#if NETMF
+                return (string)(this.GetSystemProperty(MessageSystemPropertyNames.To) ?? string.Empty);
```

Comment:
changed wherever it could be changed (won't work with non-nullable types such as DateTime)